discord/sink.py

```diff
@@ -0,0 +1,914 @@
+        The synchronization source identifier for the originator of this SR packet.
+    nts: :class:`int`
+        NTP timestamp. Indicates the wallclock time when this report was sent
+        so that it may be used in combination with timestamps returned in
+        reception reports from other receivers to measure round-trip
+        propagation to those receivers.
+    rts: :class:`int`
+        RTP timestamp. Corresponds to the same time as the NTP timestamp (above),
+        but in the same units and with the same random offset as the RTP
+        timestamps in data packets.
+    spc: :class:`int`
+        The total number of RTP data packets transmitted by the sender since
+        starting transmission up until the time this SR packet was generated.
+        The count is reset if the sender changes its SSRC identifier.
+    soc: :class:`int`
+        The total number of payload octets (i.e., not including header or padding)
+        transmitted in RTP data packets by the sender since starting transmission
+        up until the time this SR packet was generated. The count is reset if
+        the sender changes its SSRC identifier.
+    report_blocks: Sequence[:class:`RTCPReceiverReportPacket`]
+        Sequence of :class:`RTCPReceiverReportPacket` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+
+    __slots__ = RTCPPacket.__slots__ + (
+        "ssrc",
+        "nts",
+        "rts",
+        "spc",
+        "soc",
+        "report_blocks",
+        "extension",
+    )
+
+    ssrc: int
+    nts: int
+    rts: int
+    spc: int
+    soc: int
+    report_blocks: List
+    extension: bytes
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        self.ssrc, self.nts, self.rts, self.spc, self.soc = struct.unpack_from("!IQ3I", buffer=data)
+        self.report_blocks = []
+        self.extension = data[24:]
+
+
+class RTCPReceiverReportPacket(RTCPPacket):
+    """RTCP Receiver Report packet which provides quality feedback.
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/20.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    ssrc: :class:`int`
+        The synchronization source identifier for the originator of this SR packet.
+    report_blocks: Sequence[:class:`RTCPReceiverReportPacket`]
+        Sequence of :class:`RTCPReceiverReportPacket` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+
+    __slots__ = RTCPPacket.__slots__ + (
```

Comment:
This isn't how slots work, you only need to define the slots you use in this subclass

Reply:
Ah, I was unaware of this